bondcpp/src/bond.cpp

```diff
@@ -70,75 +70,87 @@ static std::string makeUUID()
 
 Bond::Bond(
   const std::string & topic, const std::string & id,
-  rclcpp_lifecycle::LifecycleNode::SharedPtr nh,
-  std::function<void(void)> on_broken,
-  std::function<void(void)> on_formed)
-: bondsm_(new BondSM(this)),
+  rclcpp::node_interfaces::NodeBaseInterface::SharedPtr node_base,
+  rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr node_logging,
+  rclcpp::node_interfaces::NodeParametersInterface::SharedPtr node_params,
+  rclcpp::node_interfaces::NodeTimersInterface::SharedPtr node_timers,
+  rclcpp::node_interfaces::NodeTopicsInterface::SharedPtr node_topics,
+  EventCallback on_broken,
+  EventCallback on_formed)
+: node_base_(node_base),
+  node_logging_(node_logging),
+  node_timers_(node_timers),
+  bondsm_(std::make_unique<BondSM>(this)),
   sm_(*bondsm_),
   topic_(topic),
   id_(id),
   instance_id_(makeUUID()),
   on_broken_(on_broken),
   on_formed_(on_formed),
-  sisterDiedFirst_(false),
-  started_(false),
-  connect_timer_reset_flag_(false),
-  disconnect_timer_reset_flag_(false),
-  deadpublishing_timer_reset_flag_(false),
-  disable_heartbeat_timeout_(false)
-{
-  node_base_ = nh->get_node_base_interface();
-  node_logging_ = nh->get_node_logging_interface();
-  node_timers_ = nh->get_node_timers_interface();
-  if (!nh->has_parameter(bond::msg::Constants::DISABLE_HEARTBEAT_TIMEOUT_PARAM)) {
-    nh->declare_parameter(bond::msg::Constants::DISABLE_HEARTBEAT_TIMEOUT_PARAM, false);
+  connect_timeout_(
+    rclcpp::Duration::from_seconds(bond::msg::Constants::DEFAULT_CONNECT_TIMEOUT)),
+  disconnect_timeout_(
+    rclcpp::Duration::from_seconds(bond::msg::Constants::DEFAULT_DISCONNECT_TIMEOUT)),
+  heartbeat_timeout_(
+    rclcpp::Duration::from_seconds(bond::msg::Constants::DEFAULT_HEARTBEAT_TIMEOUT)),
+  heartbeat_period_(
+    rclcpp::Duration::from_seconds(bond::msg::Constants::DEFAULT_HEARTBEAT_PERIOD)),
+  dead_publish_period_(
+    rclcpp::Duration::from_seconds(bond::msg::Constants::DEAD_PUBLISH_PERIOD))
+{
+  if (!node_params->has_parameter(bond::msg::Constants::DISABLE_HEARTBEAT_TIMEOUT_PARAM)) {
+    node_params->declare_parameter(
+      bond::msg::Constants::DISABLE_HEARTBEAT_TIMEOUT_PARAM,
+      rclcpp::ParameterValue(false));
   }
 
   disable_heartbeat_timeout_ =
-    nh->get_parameter(bond::msg::Constants::DISABLE_HEARTBEAT_TIMEOUT_PARAM).as_bool();
+    node_params->get_parameter(bond::msg::Constants::DISABLE_HEARTBEAT_TIMEOUT_PARAM).as_bool();
 
   setupConnections();
 
-  pub_ = rclcpp::create_publisher<bond::msg::Status>(nh, topic_, rclcpp::QoS(rclcpp::KeepLast(5)));
-  sub_ = nh->create_subscription<bond::msg::Status>(
-    topic_, rclcpp::QoS(100),
+  pub_ = rclcpp::create_publisher<bond::msg::Status>(
+    node_params,
+    node_topics,
+    topic_,
+    rclcpp::QoS(rclcpp::KeepLast(5)));
+
+  sub_ = rclcpp::create_subscription<bond::msg::Status>(
+    node_params,
+    node_topics,
+    topic_,
+    rclcpp::QoS(100),
     std::bind(&Bond::bondStatusCB, this, std::placeholders::_1));
```

Comment:
Need `#include <functional>` at the top of this file for use of `std::bind`.

(also, while you are playing with standard includes, I think `#include <iostream>` can be removed)